crates/net/network/src/cache.rs

```diff
@@ -9,12 +19,19 @@ use std::{borrow::Borrow, hash::Hash, num::NonZeroUsize};
 pub struct LruCache<T: Hash + Eq> {
     limit: NonZeroUsize,
     inner: LinkedHashSet<T>,
+    tx: Option<UnboundedSender<T>>,
```

Comment:
I would prefer this feature in a new type.
The interface of the LruCache is very small, so I think a wrapper that that has

```rust
struct LruCache(Sender|Emitter|...) {
  inner: LruCache,
  tx: UnboundedSender
}
```

would be a bit cleaner imo